src/ethereum/state_oracle/interface.py

```diff
@@ -0,0 +1,213 @@
+"""
+Abstract interface for state oracle implementations.
+"""
+
+from typing import Any, Callable, Optional, Protocol
+
+from ethereum_types.bytes import Bytes20, Bytes32
+from ethereum_types.numeric import U256
+
+# Use generic types for compatibility across forks
+Account = Any
+Address = Bytes20
+
+
+class MerkleOracle(Protocol):
```

Comment:
"Oracle" is a bit of a foreign term for non-cryptography folks. The base protocol could just be `State`.

That said, we've managed to keep protocols and most OOP out of the spec so far, so we might want to rethink how we organize this.